### PR TITLE
Add the update functionality for provisioned user

### PR DESCRIPTION
**Pull request by pamodaaw**
This PR will add the update functionality of the users that are already provisioned to the Azure AD. So that when the user profile of the user in the Identity Server is updated, those changes are synced with the Azure AD. If the dynamic memberships rules are defined in the Azure AD and configured from the Identity Server, the groups of the user may get updated accordingly. 

The following attributes are defined when provisioning the user for the first time,
* userPrincipalName
* onPremisesImmutableId
* displayName
* mailNickname
* passwordProfile

During the user profile update, passwordProfile update is not supported since it requires special permissions.

### CODE REVIEW

File: components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java [hunk 1]
```diff
@@ -177,10 +178,57 @@ protected String createUser(ProvisioningEntity provisioningEntity) throws Identi
...
+                    if (log.isDebugEnabled()) {
+                        log.debug("The response received from server : " + jsonResponse.toString());
+                    }
+                }
+            } catch (IOException | JSONException e) {
+                throw new IdentityProvisioningException("Error while executing the update operation in user " +
+                        "provisioning.", e);
+            }
+        } catch (IOException e) {
+            log.error("Error while closing HttpClient.");
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java]: Shouldn't we throw the error? Anyhow the original exception need to be recorded (at least as a debug log)

Comment (pamodaaw) [reply components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java]: Updated to print the stack trace of the exceptions

**Review by darshanasbg — APPROVED** [review]

File: components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java [hunk 2]
```diff
@@ -172,15 +173,62 @@ protected String createUser(ProvisioningEntity provisioningEntity) throws Identi
...
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_NO_CONTENT) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully updated the provisioned user with id " + provisionedUserId + " in " +
+                                "the Azure Active Directory.");
+                    }
+                } else {
+                    JSONObject jsonResponse = new JSONObject(new JSONTokener(new InputStreamReader(
+                            response.getEntity().getContent())));
+                    String errorMessage = jsonResponse.getJSONObject("error").getString("message");
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java]: Can this be null?  Better handle null value
jsonResponse.getJSONObject("error").

Comment (pamodaaw) [reply components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java]: fixed

**Review by IsuraD — APPROVED** [review]

**Pull request merged**